src/dbnode/encoding/istream.go

```diff
@@ -22,21 +22,27 @@ package encoding
 
 import (
 	"bufio"
+	"encoding/binary"
 	"io"
 	"math"
 )
 
+const (
+	numBytesInInt64 = 8
+)
+
 // istream encapsulates a readable stream.
 type istream struct {
 	r         *bufio.Reader // encoded stream
 	err       error         // error encountered
 	current   byte          // current byte we are working off of
+	buffer    []byte        // buffer for reading in multiple bytes
 	remaining uint          // bits remaining in current to be read
 }
 
 // NewIStream creates a new Istream
 func NewIStream(reader io.Reader, bufioSize int) IStream {
-	return &istream{r: bufio.NewReaderSize(reader, bufioSize)}
+	return &istream{r: bufio.NewReaderSize(reader, bufioSize), buffer: make([]byte, numBytesInInt64, numBytesInInt64)}
```

Comment:
I think this is much clearer as `buffer: make([]byte, 8)`, number of bytes in int64 is not going to change :p

Reply:
Yeah that's true. I was mainly trying to distinguish between 8 bits in a byte vs 8 bytes in a uint64, since we are using both purposes of 8 in this file. But agreed it probably is pretty clear based on the context so I'll just use 8 everywhere.